app/src/main/java/com/android/shelfLife/ui/newoverview/NewHouseHoldCreationScreen.kt

```diff
@@ -144,143 +121,112 @@ fun HouseHoldCreationScreen(
+                          Icon(
+                              imageVector = Icons.Default.Add,
+                              contentDescription = "Add Email",
+                              tint = MaterialTheme.colorScheme.onSecondaryContainer)
+                        }
+                  }
 
-                FloatingActionButton(
-                    modifier = Modifier.padding(top = 20.dp, bottom = 20.dp).testTag("AddEmailFab"),
-                    onClick = { showEmailTextField = !showEmailTextField },
-                    containerColor = MaterialTheme.colorScheme.secondaryContainer,
-                ) {
-                  Icon(
-                      imageVector = Icons.Default.Add,
-                      contentDescription = "Add Email",
-                      tint = MaterialTheme.colorScheme.onSecondaryContainer)
-                }
-              }
-              // Confirm and Cancel buttons
               CustomButtons(
                   button1OnClick = { navigationActions.goBack() },
                   button1TestTag = "CancelButton",
                   button1Text = stringResource(R.string.cancel_button),
                   button2OnClick = {
-                    if (householdCreationScreenViewModel.newHouseholdNameIsInvalid(houseHoldName)) {
-                      isError = true
-                    } else {
-                      if (householdToEdit != null) {
-                        val updatedHouseHold = householdToEdit!!.copy(name = houseHoldName)
-                        householdCreationScreenViewModel.getUserIdsByEmails(
-                            memberEmailList,
-                            callback = { emailToUserIds ->
-                              if (emailToUserIds.isNotEmpty()) {
-                                val oldUidList = updatedHouseHold.members
-                                val uidList = memberEmailList.map { emailToUserIds[it]!! }
-                                if (oldUidList.size < uidList.size) {
-                                  householdCreationScreenViewModel.updateHousehold(
-                                      householdToEdit!!.copy(
-                                          name = houseHoldName, members = uidList),
-                                      false)
-                                } else if (oldUidList.size > uidList.size) {
-                                  householdCreationScreenViewModel.updateHousehold(
-                                      householdToEdit!!.copy(
-                                          name = houseHoldName, members = uidList),
-                                      true)
-                                }
-                                householdCreationScreenViewModel.updateHousehold(updatedHouseHold)
-                              }
-                            })
+                    coroutineScope.launch {
+                      val success = viewModel.confirmHouseholdActions(houseHoldName)
```

Comment:
Yeah i think this is very elegant; once again I see that this whole logic is done within the ViewModel